runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/TimerTest.java

```diff
@@ -36,39 +38,121 @@
   private static final Instant INSTANT = Instant.now();
 
   @Test
-  public void testTimer() {
-    Timer<Void> timerA = Timer.of(INSTANT);
-    assertEquals(INSTANT, timerA.getTimestamp());
-    assertNull(timerA.getPayload());
+  public void testClearTimer() {
+    Timer<String> clearedTimer = Timer.cleared("timer", "tag");
+    assertTrue(clearedTimer.getClearBit());
+    assertEquals("timer", clearedTimer.getUserKey());
+    assertEquals("tag", clearedTimer.getDynamicTimerTag());
+  }
 
-    Timer<String> timerB = Timer.of(INSTANT, "ABC");
-    assertEquals(INSTANT, timerB.getTimestamp());
-    assertEquals("ABC", timerB.getPayload());
+  @Test
+  public void testTimer() {
+    Timer<String> timer =
+        Timer.of(
+            "key",
+            "tag",
+            INSTANT,
```

Comment:
use two different instant objects otherwise you can't tell if they get swapped accidentally